src/app/parcels/form/ParcelForm.tsx

```diff
@@ -174,6 +184,44 @@ const ParcelForm: React.FC<ParcelFormProps> = ({
         }
     }, [clientDetails, clientIdForFetch]);
 
+    useEffect(() => {
+        const getTimeSlots = async (): Promise<void> => {
+            if (fields.collectionCentre) {
+                const { data, error } = await getActiveTimeSlotsForCollectionCentre(
+                    fields.collectionCentre,
+                    supabase
+                );
+
+                if (error) {
+                    let errorMessage;
+                    switch (error.type) {
+                        case "collectionTimeSlotsFetchFailed":
+                            errorMessage = "Failed to fetch collection time slots";
+                            break;
+                    }
+                    setSubmitErrorMessage(`${errorMessage}. Log ID: ${error.logId}`);
+                    return;
+                }
```

Comment:
Don't know if you can see it but github seems to be complaining: `Type '"noCollectionTimeSlotsData"' is not comparable to type '"collectionTimeSlotsFetchFailed"'.`
Not sure what's wrong with it but thought I'll bring attention to it

Reply:
I've removed "noCollectionTimeSlotsData" from the return type getActiveTimeSlotsForCollectionCentre but forgot to remove it from the switch - it's passing now :)